mi/dataset/driver/zplsc_c/zplsc_echogram_generator.py

```diff
@@ -266,61 +275,62 @@ def get_data_filenames(self, date_dirs_url, data_date):
+        This method will aggregate the 24 1-hour files residing at the path
+        passed in for the date passed in.  It will copy the 24 files to a
+        temporary directory and then aggregate them and store the single file
+        in the temporary directory.  It will return the file name of the
+        aggregated file and the file name of the echogram file that will be
         generated.
 
-        This method will raise an exception if there is an issue creating
-        the local ZPLSC Echogram directory.
+        Exceptions raised This method will raise an exception if there is an
+        issue creating the local ZPLSC Echogram directory.
 
-        :param date_dirs_url: The URL of the raw data server where the 24 1-hour files reside.
+        :param date_dirs_path: The path of the raw data server where the 24 1-hour files reside.
         :param data_date: The date of the raw data for the echogram to be generated.
         :return: zplsc_24_datafile: The 24-hour concatenated raw data file
                  zplsc_echogram_file_path: The file path for the echogram to be generated.
         """
 
         zplsc_echogram_file_path = None
 
-        filenames_list, raw_data_url, raw_datafile_prefix = self.get_data_filenames(date_dirs_url, data_date)
+        filenames_list, raw_data_path, raw_datafile_prefix = self.get_data_filenames(date_dirs_path, data_date)
         if len(filenames_list) != 24:
             return '', None
 
-        # Download the 24 1-hour raw data files to a temporary local directory.
+        # Copy the 24 1-hour raw data files to a temporary local directory.
         for raw_data_file in filenames_list:
-            remote_raw_data_file = os.path.join(raw_data_url, raw_data_file)
+            remote_raw_data_file = os.path.join(raw_data_path, raw_data_file)
             local_raw_data = os.path.join(self.temp_directory, raw_data_file)
             try:
-                urllib.urlretrieve(remote_raw_data_file, local_raw_data)
-            except urllib.ContentTooShortError as ex:
-                log.error('Error retrieving: %s: %s', remote_raw_data_file, ex.message)
+                shutil.copyfile(remote_raw_data_file, local_raw_data)
+            except OSError as ex:
+                log.error('Error copying data file to temporary directory: %s: %s', remote_raw_data_file, ex.message)
                 continue
 
         # Concatenate the 24 1-hour raw data files to 1 24-hour raw data file and return the filename.
         zplsc_24_datafilename = self.zplsc_24_datafile_prefix + raw_datafile_prefix
         zplsc_24_datafile = os.path.join(self.temp_directory, zplsc_24_datafilename) + RAW_FILE_EXT
-        os.system('cat ' + os.path.join(self.temp_directory, raw_datafile_prefix) + '*' + RAW_FILE_EXT + ' > ' +
-                  zplsc_24_datafile)
+        raw_data_glob = os.path.join(raw_data_path, raw_datafile_prefix) + '*' + RAW_FILE_EXT
+        os.system('cat ' + raw_data_glob + ' > ' + zplsc_24_datafile)
```

Comment:
not happy with the system command, but since this was a pre-existing condition I suppose I have to let it slide...

Reply:
:)